opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/services/ServiceContextProvider.java

```diff
@@ -72,24 +85,60 @@
     @NotNull
     @SneakyThrows
     private <T extends FacadeServiceableGetter> ServiceSession findServiceSessionByIdOrCreate(T request) {
-        if (null != request.getFacadeServiceable().getServiceSessionId()) {
-            Optional<ServiceSession> existingSession = serviceSessions
-                    .findById(request.getFacadeServiceable().getServiceSessionId());
-            if (existingSession.isPresent()) {
-                return existingSession.get();
+        FacadeServiceableRequest facadeServiceable = request.getFacadeServiceable();
+        String sessionPassword = facadeServiceable.getSessionPassword();
+
+        Optional<ServiceSession> existingSession = Optional.empty();
+        if (null != facadeServiceable.getServiceSessionId()) {
+            existingSession = serviceSessions.findById(facadeServiceable.getServiceSessionId());
+        }
+
+        byte[] salt = getNewSalt();
```

Comment:
Suggest to move decryption stuff to separate function

Reply:
ok